pkg/controller/servicemesh/memberroll/namespace_reconciler.go

```diff
@@ -167,7 +167,7 @@ func (r *namespaceReconciler) removeNamespaceFromMesh(ctx context.Context, names
 		for _, rb := range rbList.Items {
 			logger.Info("deleting RoleBinding for mesh ServiceAccount", "RoleBinding", rb.GetName())
 			err = r.Client.Delete(ctx, &rb)
-			if err != nil {
+			if err != nil && !(apierrors.IsNotFound(err) || apierrors.IsGone(err)) {
```

Comment:
Do we need to reset `err` to `nil` here?  I ask because it looks like it's declared out of this scope and might be read elsewhere.

Reply:
No, as it gets overwritten in the next statement.